shared/js/url_helper.js

```diff
@@ -4,13 +4,8 @@ var rscheme = /^(?:[a-z\u00a1-\uffff0-9-+]+)(?::(?:\/\/)?)/i;
 
 var UrlHelper = {
 
-  // Placeholder anchor tag to format URLs.
-  a: null,
-
   getUrlFromInput: function urlHelper_getUrlFromInput(input) {
-    this.a = this.a || document.createElement('a');
-    this.a.href = input;
-    return this.a.href;
+    return new URL(input).href;
```

Comment:
Are you sure that deals correctly with 'input' being a relative url? Don't we need to resolve against the document url?